src/megaapi_impl.cpp

```diff
@@ -11774,9 +11774,15 @@ MegaNodeList* MegaApiImpl::search(MegaNode *n, const char* searchString, MegaCan
         if (target == MegaApi::SEARCH_TARGET_OUTSHARE)
         {
             // Search on outshares
+            std::set<MegaHandle>outsharesHandles;
             unique_ptr<MegaShareList>shares (getOutShares(MegaApi::ORDER_NONE));
             for (int i = 0; i < shares->size() && !(cancelToken && cancelToken->isCancelled()); i++)
             {
+                if (outsharesHandles.find(shares->get(i)->getNodeHandle()) != outsharesHandles.end())
+                {
+                    continue;   // avoid duplicates
+                }
+                outsharesHandles.insert(shares->get(i)->getNodeHandle());
```

Comment:
You can improve readability a bit:
```suggestion cpp
                handle h = shares->get(i)->getNodeHandle();
                if (outsharesHandles.find(h) != outsharesHandles.end())
                {
                    // shares list includes an item per outshare AND per sharee/user
                    continue;   // avoid duplicates
                }
                outsharesHandles.insert(h);
```
Also, should use the declared `h` in the next line too :)